packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -1546,7 +1546,7 @@ class RenderFittedBox extends RenderProxyBox {
     if (_transform != null)
       return;
 
-    if (child == null) {
+    if (child == null || size.isEmpty) {
       _hasVisualOverflow = false;
```

Comment:
This doesn't seem right. Surely if the size is empty you're _more_ likely to have overflow.